Source/NujasAI/Public/NujasAIGlobals.h

```diff
@@ -0,0 +1,38 @@
+// Copyright 2018-2019 The Janus Project | 2034 Complex LLC. All Rights Reserved.
+
+#pragma once
+
+#include "Interface.h"
+#include "NujasAIGlobals.generated.h"
+
+UINTERFACE(Blueprintable)
+class NUJASAI_API UAggressive : public UInterface
+{
+	GENERATED_BODY()
+};
+
+class IAggressive
+{
+	GENERATED_BODY()
+
+public:
+	// callback for when the actor turns aggressive
+	UFUNCTION(BlueprintImplementableEvent, Category = "Aggressive")
+	bool OnTurAgressive();
+
+	// callback when the actor is deemed none-aggressive
+	UFUNCTION(BlueprintImplementableEvent, Category = "Aggressive")
+	bool OnTurnCalm();
+
+	// helper func to determine the AI's priority. The priority is usually set manually depending on the AI
```

Comment:
This value is also set based on distance. i.e the designer could say that a draugr's priority at 5 meters away from the target is 10 but at 100 meters the priority is 1.

Reply:
That value is determined on the Bot's side. This function just returns it